ydb/core/change_exchange/change_sender_common_ops.cpp

```diff
@@ -162,6 +162,14 @@ void TBaseChangeSender::SendRecords() {
     bool needToResolve = false;
 
     while (it != PendingSent.end()) {
+        if (Enqueued && Enqueued.begin()->Order <= it->first) {
```

Comment:
I'm worried this may deadlock. When we previously sent records we removed them from `MemUsage`, and we might have loaded more records in the available space. Now we have reenqueued some records, but they might not be able to start loading, because `MemUsage` no longer has enough space. This would deadlock: we don't send ready records because we have some enqueued, and we don't load enqueued records because we don't have enough memory.